src/aiida/manage/configuration/profile.py

```diff
@@ -29,41 +31,44 @@
 class Profile:
     """Class that models a profile as it is stored in the configuration file of an AiiDA instance."""
 
-    KEY_UUID = 'PROFILE_UUID'
-    KEY_DEFAULT_USER_EMAIL = 'default_user_email'
-    KEY_STORAGE = 'storage'
-    KEY_PROCESS = 'process_control'
-    KEY_STORAGE_BACKEND = 'backend'
-    KEY_STORAGE_CONFIG = 'config'
-    KEY_PROCESS_BACKEND = 'backend'
-    KEY_PROCESS_CONFIG = 'config'
-    KEY_OPTIONS = 'options'
-    KEY_TEST_PROFILE = 'test_profile'
+    KEY_UUID: str = 'PROFILE_UUID'
+    KEY_DEFAULT_USER_EMAIL: str = 'default_user_email'
+    KEY_STORAGE: str = 'storage'
+    KEY_PROCESS: str = 'process_control'
+    KEY_STORAGE_BACKEND: str = 'backend'
+    KEY_STORAGE_CONFIG: str = 'config'
+    KEY_PROCESS_BACKEND: str = 'backend'
+    KEY_PROCESS_CONFIG: str = 'config'
+    KEY_OPTIONS: str = 'options'
+    KEY_TEST_PROFILE: str = 'test_profile'
 
     # keys that are expected to be in the parsed configuration
-    REQUIRED_KEYS = (
+    REQUIRED_KEYS: tuple[str, str] = (
         KEY_STORAGE,
         KEY_PROCESS,
     )
 
-    def __init__(self, name: str, config: Mapping[str, Any], validate=True):
+    def __init__(
+        self, name: str, config: abc.Mapping[str, Any], config_folder: pathlib.Path | None = None, validate: bool = True
+    ):
         """Load a profile with the profile configuration."""
-        if not isinstance(config, collections.abc.Mapping):
-            raise TypeError(f'config should be a mapping but is {type(config)}')
+        _ = type_check(config, abc.Mapping)
```

Comment:
I imagine this change is merely there to satisfy your personal linter? If so, I would prefer to remove it. I don't think there is anything wrong with ignoring a return value of a function call. A linter forcing us to assign it to an anonymous variable to never be used is just counter-productive.

Reply:
Yep, removed.